go/vt/vtgate/planbuilder/system_tables.go

```diff
@@ -151,10 +160,63 @@ func extractInfoSchemaRoutingPredicate(in sqlparser.Expr) ([]evalengine.Expr, []
 			cmp.Right = populateValTuple(valTuples, len(colNames))
 			return sysTableSchemas, sysTableNames, nil
 		}
+	case *sqlparser.OrExpr:
+		sysTableSchemas := make([]evalengine.Expr, 0)
+		sysTableNames := make([]evalengine.Expr, 0)
+		schemas, names, err := extractInfoSchemaRoutingPredicate(cmp.Left, sysTableSchema, sysTableName)
+		if err != nil {
+			return nil, nil, err
+		}
+		sysTableSchemas = append(sysTableSchemas, schemas...)
```

Comment:
this code could be simplified as 

sysTableSchema = append(sysTableSchema, schemas)